contentcuration/contentcuration/frontend/channelEdit/views/files/thumbnails/ThumbnailGenerator.vue

```diff
@@ -0,0 +1,203 @@
+<template>
+
+  <Uploader ref="uploader" :presetID="presetID">
+    <Alert
+      ref="error"
+      :header="$tr('thumbnailGenerationFailedHeader')"
+      :text="$tr('thumbnailGenerationFailedText')"
+    />
+    <slot :generate="generate"></slot>
+  </Uploader>
+
+</template>
+
+<script>
+
+  import { mapActions } from 'vuex';
+  import chunk from 'lodash/chunk';
+  import min from 'lodash/min';
+  import map from 'lodash/map';
+  import max from 'lodash/max';
+  import pdfJSLib from 'pdfjs-dist';
+  import Alert from 'edit_channel/sharedComponents/Alert.vue';
+  import Uploader from 'frontend/channelEdit/views/files/Uploader';
+
+  const MAX_AUDIO_SAMPLE_SIZE = 64000;
+
+  export default {
+    name: 'ThumbnailGenerator',
+    components: {
+      Alert,
+      Uploader,
+    },
+    props: {
+      width: {
+        type: Number,
+        default: 320,
+      },
```

Comment:
@jayoshih Note in `pressurecooker` the default size for generated thumbnails is `400x225`
https://github.com/learningequality/pressurecooker/blob/master/pressurecooker/images.py#L33

It seems `320x180` is a better choice for default since both dimensions are round. It might lead to smaller image sizes too...

Note the default thumbnail size  on Kolibri is `210x118.125` (on desktop viewport) so either choice is plenty, although some minor smudging will happen.

Reply:
ooh nice catch!